python/nammu/SOAPClient/HTTPRequest.py

```diff
@@ -14,39 +16,121 @@ def __init__(self, url, method, **kwargs):
         if method == 'POST':
             if 'command' in kwargs.keys():
                 self.create_request_message(kwargs['command'], kwargs['keys'],
-                                            kwargs['attachment'])
+                                            kwargs['atf_basename'],
+                                            kwargs['atf_text'])
             else:
                 self.create_response_message(kwargs['keys'])
 
-    def create_request_message(self, command, keys, attachment):
+    def create_request_message(self, command, keys, atf_basename, atf_text):
         """
         Send attachment to server containing ATF file and necessary data to
         run given command (validate, lemmatise, etc).
         """
-        self.create_soap_envelope(command=command,
-                                  keys=keys,
-                                  attachment=attachment)
+        self.mtompkg = MIMEMultipart('related')
+        self.set_multipart_params()
+        self.set_soap_envelope(command=command,
+                               keys=keys,
+                               atf_basename=atf_basename,
+                               atf_text=atf_text)
+        self.rootpkg = MIMEApplication(self.envelope, 'xop+xml', encode_7or8bit)
+        self.set_multipart_payload()
+        self.document = MIMEBase('*','*')
+        self.set_document_payload(atf_basename, atf_text)
+
+        #The headers can't be created until the body is finished since they need
+        #it to populate the Content-Length header
+        self.set_multipart_headers()
+
+        print "*"*30
```

Comment:
I guess this should be replaced by a logging functionality but that can wait till after this PR